alicloud/resource_alicloud_emr_cluster.go

```diff
@@ -0,0 +1,410 @@
+				}
+
+				if v, ok := kv["sys_disk_capacity"]; ok {
+					hostGroup.SysDiskCapacity = v.(string)
+				}
+
+				if v, ok := kv["disk_capacity"]; ok {
+					hostGroup.DiskCapacity = v.(string)
+				}
+
+				if v, ok := kv["sys_disk_type"]; ok {
+					hostGroup.SysDiskType = v.(string)
+				}
+
+				if v, ok := kv["disk_type"]; ok {
+					hostGroup.DiskType = v.(string)
+				}
+
+				if v, ok := kv["host_group_name"]; ok {
+					hostGroup.HostGroupName = v.(string)
+				}
+
+				if v, ok := kv["vswitch_id"]; ok {
+					hostGroup.VSwitchId = v.(string)
+				}
+
+				if v, ok := kv["disk_count"]; ok {
+					hostGroup.DiskCount = v.(string)
+				}
+
+				if v, ok := kv["auto_renew"]; ok {
+					hostGroup.AutoRenew = v.(string)
+				}
+
+				if v, ok := kv["gpu_driver"]; ok {
+					hostGroup.GpuDriver = v.(string)
+				}
+
+				if v, ok := kv["node_count"]; ok {
+					hostGroup.NodeCount = v.(string)
+				}
+
+				if v, ok := kv["instance_type"]; ok {
+					hostGroup.InstanceType = v.(string)
+				}
+
+				if v, ok := kv["charge_type"]; ok {
+					hostGroup.ChargeType = v.(string)
+				}
+
+				if v, ok := kv["host_group_type"]; ok {
+					hostGroup.HostGroupType = v.(string)
+				}
+
+				hostGroups = append(hostGroups, hostGroup)
+			}
+		}
+		request.HostGroup = &hostGroups
+
+		raw, err := client.WithEmrClient(func(emrClient *emr.Client) (interface{}, error) {
+			return emrClient.CreateClusterV2(request)
+		})
+
+		if err != nil {
+			return resource.NonRetryableError(err)
+		}
+
+		response, _ := raw.(*emr.CreateClusterV2Response)
+		d.SetId(response.ClusterId)
+		return nil
+	}); err != nil {
+		return WrapErrorf(err, DefaultErrorMsg, "alicloud_emr_flow_project", "CreateFlowProject", AliyunLogGoSdkERROR)
+	}
+	return nil
+}
+
+func resourceAlicloudEmrClusterRead(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*connectivity.AliyunClient)
+	request := emr.CreateDescribeClusterV2Request()
+	request.Id = d.Id()
+	object, err := client.WithEmrClient(func(emrClient *emr.Client) (interface{}, error) {
+		return emrClient.DescribeClusterV2(request)
+	})
+
+	if err != nil {
+		if NotFoundError(err) {
+			d.SetId("")
+			return nil
+		}
+		return WrapError(err)
+	}
+	addDebug("clusterInfo", object)
+	d.Set("clusterInfo", object)
+	return nil
+}
+
+func resourceAlicloudEmrClusterUpdate(d *schema.ResourceData, meta interface{}) error {
+	return nil
+}
+
+func resourceAlicloudEmrClusterDelete(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*connectivity.AliyunClient)
+
+	if err := resource.Retry(2*time.Minute, func() *resource.RetryError {
+		request := emr.CreateReleaseClusterRequest()
+
+		request.Id = d.Id()
+		request.ForceRelease = requests.NewBoolean(true)
+
+		raw, err := client.WithEmrClient(func(emrClient *emr.Client) (interface{}, error) {
+			return emrClient.ReleaseCluster(request)
+		})
+
+		if err != nil {
+			return resource.NonRetryableError(err)
+		}
+
+		addDebug("DeleteCluster", raw)
```

Comment:
addDebug uniformly modified to addDebug(request.GetActionName(), raw, request.RpcRequest, request)